File: 프로그래머스/3/12907. 거스름돈/거스름돈.js
function solution(n, money) {
    
    var dp = Array(n+1).fill(0);
    
    dp[0] = 1;

    for(let i=0; i<money.length; i++){
        var now = money[i];
        for(let j=now; j<n+1; j++){
            dp[j] += dp[j-now];
            dp[j] %= 1000000007;
        }
    }

    return dp[n];
}

File: 프로그래머스/3/12907. 거스름돈/거스름돈.java
class Solution {
    public int solution(int n, int[] money) {
        
        long dp[] = new long[n+1];
        dp[0] = 1;
        
        for(int i=0; i<money.length; i++){
            int now = money[i];
            for(int j=now; j<=n; j++){
                dp[j] += dp[j-now];
                dp[j] %= 1000000007;
            }
        }
        
        return (int)dp[n];
    }
}